packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -177,17 +182,59 @@ export class Toolkit extends CloudAssemblySourceBuilder implements AsyncDisposab
             usePreviousParameters: parameters?.keepExistingParameters,
           },
         );
+
+        const result: EnvironmentBootstrapResult = {
+          environment: environment.name,
+          status: bootstrapResult.noOp ? 'no-op' : 'success',
+          message: bootstrapResult.noOp ? 'No changes required' : 'Successfully bootstrapped',
+          duration: Date.now() - envStartTime,
+        };
+
+        results.push(result);
+
         const message = bootstrapResult.noOp
           ? ` ✅  ${environment.name} (no changes)`
           : ` ✅  ${environment.name}`;
 
         await ioHelper.notify(IO.CDK_TOOLKIT_I9900.msg(chalk.green('\n' + message), { environment }));
         await bootstrapSpan.end();
       } catch (e: any) {
+        const result: EnvironmentBootstrapResult = {
+          environment: environment.name,
+          status: 'failed',
+          error: e,
+          message: formatErrorMessage(e),
+          duration: Date.now() - envStartTime,
+        };
+
+        results.push(result);
+
         await ioHelper.notify(IO.CDK_TOOLKIT_E9900.msg(`\n ❌  ${chalk.bold(environment.name)} failed: ${formatErrorMessage(e)}`, { error: e }));
         throw e;
       }
     })));
+
+    const summary = results.reduce((acc, result) => {
+      acc.total++;
+      switch (result.status) {
+        case 'success':
+          acc.successful++;
+          break;
+        case 'failed':
+          acc.failed++;
+          break;
+        case 'no-op':
+          acc.noOp++;
+          break;
+      }
+      return acc;
+    }, { total: 0, successful: 0, failed: 0, noOp: 0 });
+
+    return {
+      environments: results,
+      summary,
```

Comment:
I think let's just loose the summary. If we return, it's successful. If someone really cares about the counts for success vs. no-op, they can do that themselves.

Reply:
Yup, makes sense. Removed the summary.